src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.AuthorizeNet.Web.Managers;
+using VirtoCommerce.OrdersModule.Core.Model;
+using VirtoCommerce.OrdersModule.Core.Services;
+using VirtoCommerce.PaymentModule.Core.Model.Search;
+using VirtoCommerce.PaymentModule.Core.Services;
+using VirtoCommerce.PaymentModule.Model.Requests;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.StoreModule.Core.Services;
+
+namespace VirtoCommerce.AuthorizeNet.Web.Services
+{
+    public class AuthorizeNetRegisterPaymentService : IAuthorizeNetRegisterPaymentService
```

Comment:
Why do we need this separated abstraction, is it impossible to move into PaymentMethod?

Reply:
It defines payment result handling workflow. 
It was extracted from the old controller to facilitate controller-contained logic:
https://github.com/VirtoCommerce/vc-module-Authorize.Net/pull/5/files/6f5afbab40be3ff01eff211e68743dc1d900e594#diff-9a6d5d29037c4d28b6805d97050723afL33

Not sure we need to move it to PaymentMethod, as it has the methods for specific actions (Process, PostProcess, Refund etc.). New abstraction is like a manager calling different PaymentMethod's operations.
Almost the same code excluding AuthorizeNet specific and return type exists in Orders module:
https://github.com/VirtoCommerce/vc-module-order/blob/release/3.0.0/src/VirtoCommerce.OrdersModule.Web/Controllers/Api/OrderModuleController.cs#L419

So I think we could have some abstraction for this logic to use it in all places.

Reply:
Ok, then. Anyway, we will change this logic soon into an absolutely different manner.